onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -843,21 +993,38 @@ common::Status TensorrtExecutionProvider::Provider_Compile(const std::vector<onn
       }
     }
 
+    // Load calibration table for INT8 quantization
+    std::unordered_map<std::string, float> dynamic_range_map;
+    if (int8_enable_ && trt_builder->platformHasFastInt8()) {
+      const std::string calibration_cache_path = GetCachePath(cache_path_, int8_calibration_cache_name_);
+      if (!ReadDynamicRange(calibration_cache_path, int8_use_tensorrt_calibration_table_, dynamic_range_map)) {
+        throw std::runtime_error("Failed to read INT8 calibration table " + calibration_cache_path);
+      }
+    }
```

Comment:
should we have some warning logs for if the user tries to enable int8 or fp16 on hardware that does not support it?